UI/src/model/selectedLaneModelling.js

```diff
@@ -0,0 +1,588 @@
+          return prop.publicId == "lane_code";
+        }).values).value;
+
+        var properties = _.filter(self.getValue(currentLaneNumber), function(property){ return property.publicId !== currentPropertyValue.publicId; });
+        properties.push(currentPropertyValue);
+        self.setValue(currentLaneNumber, {properties: properties});
+      });
+    };
+
+    this.getValue = function(laneNumber, marker) {
+      var lane = self.getLane(laneNumber, marker);
+
+      var value = getProperty(lane, 'properties');
+      return value;
+    };
+
+    this.getModifiedBy = function() {
+      return dateutil.extractLatestModifications(selection, 'modifiedAt').modifiedBy;
+    };
+
+    this.getModifiedDateTime = function() {
+      return dateutil.extractLatestModifications(selection, 'modifiedAt').modifiedAt;
+    };
+
+    this.getCreatedBy = function(laneNumber) {
+      return getProperty(self.getLane(laneNumber), 'createdBy');
+    };
+
+    this.getCreatedDateTime = function(laneNumber) {
+      return getProperty(self.getLane(laneNumber), 'createdAt');
+    };
+
+    this.getAdministrativeClass = function(laneNumber) {
+      var value = getProperty(self.getLane(laneNumber), 'administrativeClass');
+      return _.isNull(value) ? undefined : value;
+    };
+
+    this.getVerifiedBy = function(laneNumber) {
+      return getProperty(self.getLane(laneNumber), 'verifiedBy');
+    };
+
+    this.getVerifiedDateTime = function(laneNumber) {
+      return getProperty(self.getLane(laneNumber), 'verifiedAt');
+    };
+
+    this.get = function() {
+      return selection;
+    };
+
+    this.count = function() {
+      return selection.length;
+    };
+
+    this.setNewLane = function(laneNumber) {
+      var newLane;
+      if(laneNumber.toString()[1] == 2){
+        newLane = _.cloneDeep(self.getLane(laneNumber-1));
+      }else{
+        newLane = _.cloneDeep(self.getLane(laneNumber-2));
+      }
+
+      var outerLaneIsMainLane = laneNumber.toString()[1] == 2 || laneNumber.toString()[1] == 3;
+
+      var properties = _.filter(newLane.properties, function (property) {
+        if(outerLaneIsMainLane)
+          return property.publicId != "lane_code" && property.publicId != "lane_type";
+
+        return property.publicId != "lane_code";
+      });
+
+      var laneCodeProperty = {publicId: "lane_code", propertyType: "read_only_number", required: "required", values: [{value: laneNumber}]};
+      properties.push(laneCodeProperty);
+      newLane.properties = properties;
+
+      newLane.id = 0;
+      selection.push(newLane);
+      dirty = true;
+    };
+
+    this.removeLane = function(laneNumber, marker) {
+        var laneIndex = _.findIndex(selection, function (lane) {
+          return (_.isEmpty(marker) || lane.marker == marker) && _.find(lane.properties, function (property) {
+            return property.publicId == "lane_code" && _.head(property.values).value == laneNumber;
+          });
+        });
```

Comment:
this is reused in setValue and expireLane below so maybe we should move it to a local function

Reply:
You are right.
Created a function for this redundant code